File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <queue>

using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
        
    priority_queue<int,vector<int>,greater<int>> pq;
    
    for (int i=0;i<scoville.size();i++){
        pq.push(scoville[i]);
    }
        
    while(pq.top() < K) {
        if (pq.size() < 2) return -1;
        
        int newScoville = pq.top();
        pq.pop();
        newScoville += pq.top() * 2;
        pq.pop();

        pq.push(newScoville);
        
        answer++;
    }
    
    return answer;
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for (int s : scoville) {
            pq.offer(s);
        }
        
        int mixCount = 0;
        
        while(pq.size() >= 2 && pq.peek() < K) {
            int first = pq.poll();
            int second = pq.poll();
            
            int newScoville = first + (second * 2);
            pq.offer(newScoville);
            mixCount++;
        }
        
        if (pq.peek() < K) return -1;
        
        return mixCount;
    }
}